src/components/popup/tx-item.tsx

```diff
@@ -126,6 +101,120 @@ const SpeedUpButton = ({
   );
 };
 
+const TxTransfers: React.FC<{ transaction: AddressTransactionWithTransfers & BoxProps }> = ({
+  transaction,
+  ...rest
+}) => {
+  return (
+    <>
+      {transaction.stx_transfers.map((stxTransfer, index) => (
+        <StxTransferItem stxTransfer={stxTransfer} parentTx={transaction} {...rest} key={index} />
+      ))}
+    </>
+  );
+};
+
+const TxItemRow: React.FC<{
+  title: string;
+  value: number | string | null;
+  caption?: string;
+  icon: JSX.Element;
+  onClick?: () => Window | null;
+}> = ({ title, caption, value, icon, onClick, ...rest }) => {
+  const [component, bind] = usePressable(true);
+  return (
+    <Box position="relative" cursor="pointer" {...bind} {...rest}>
+      <Stack
+        alignItems="center"
+        spacing="base-loose"
+        isInline
+        position="relative"
+        zIndex={2}
+        onClick={onClick}
+      >
+        {icon}
+        <SpaceBetween flexGrow={1}>
+          <Stack spacing="base-tight">
+            <Title as="h3" fontWeight="normal">
+              {title}
+            </Title>
+            <Stack isInline flexWrap="wrap">
+              <Caption variant="c2">{caption}</Caption>
+            </Stack>
+          </Stack>
+          <Stack alignItems="flex-end" spacing="base-tight">
+            {value && (
+              <Title as="h3" fontWeight="normal">
+                {value}
+              </Title>
+            )}
+          </Stack>
+        </SpaceBetween>
+      </Stack>
+      {component}
+    </Box>
+  );
+};
+
+const StxTransferItem: React.FC<{
+  stxTransfer: StxTransfer;
+  parentTx: AddressTransactionWithTransfers;
+}> = ({ stxTransfer, parentTx }) => {
+  const currentAccount = useCurrentAccount();
+  const { handleOpenTxLink } = useExplorerLink();
+  const title = 'Stacks Token Transfer';
+  const caption = getTxCaption(parentTx.tx) ?? '';
+  const isOriginator = stxTransfer.sender === currentAccount?.address;
+
+  const value = `${isOriginator ? '-' : ''}${stacksValue({
+    value: stxTransfer.amount,
+    withTicker: false,
+  })}`;
+
+  const icon = isOriginator ? IconArrowUp : IconArrowDown;
+
+  const iconWrapper = (
+    <Circle position="relative" size="36px" bg={color('accent')} color={color('bg')}>
+      <Box as={StxIcon} />
+      <TypeIconWrapper icon={icon} bg={'brand'} />
+    </Circle>
+  );
+
+  return (
+    <TxItemRow
+      title={title}
+      caption={caption}
+      value={value}
+      onClick={() => handleOpenTxLink(parentTx.tx.tx_id)}
+      icon={iconWrapper}
+    />
+  );
+};
+
+export const TxWithTransferItem: React.FC<TxWithTransferItemProps & BoxProps> = ({
```

Comment:
This component, and its corresponding interface, has confusing names.

It isn't necessarily a TxWithTransferItem, it may well be a regular transaction. If we're building a component that supports both, I would omit this information altogether. This is just the component that displays a transaction.

Reply:
Changed to TxView (TxItem is still used elsewhere)